test/com/puppetlabs/puppetdb/test/scf/storage.clj

```diff
@@ -259,6 +283,99 @@
                  :environment_id (environment-id "DEV")}]
                (query-to-vec "SELECT certname, environment_id FROM factsets")))))))
 
+(deftest fact-path-value-gc
```

Comment:
Can you grab the commit from [here](https://github.com/puppetlabs/puppetdb/pull/1246) that originally reproduced the fact path issue and add it here? Would also be easy to then run the gc and check that it "cleans up" after the concurrent tx issue.